core/include/gnuradio-4.0/HistoryBuffer.hpp

```diff
@@ -100,23 +113,176 @@ class HistoryBuffer {
     /**
      * @brief Adds a range of elements the end expiring the oldest elements beyond the buffer's capacities.
      */
-    template<typename Iter>
-    constexpr void push_back_bulk(Iter cbegin, Iter cend) noexcept {
+    template<std::input_iterator Iter>
+    constexpr void push_back(Iter cbegin, Iter cend) noexcept {
         const auto nSamplesToCopy = std::distance(cbegin, cend);
         Iter       optimizedBegin = static_cast<std::size_t>(nSamplesToCopy) > _capacity ? std::prev(cend, static_cast<std::ptrdiff_t>(_capacity)) : cbegin;
         for (auto it = optimizedBegin; it != cend; ++it) {
             push_back(*it);
         }
     }
 
+    template<std::input_iterator Iter>
```

Comment:
Why a separate name for the new push_back?

IMO, it should be named `append_range` as in `std::vector`, though that one takes a range and not a pair of iterators. We could have `append_range(begin, end)` and `append_range(range)` -- if the functions below also become `append_range`?

Reply:
I didn't want to break existing code, thus keeping the "old" `push_back_bulk(...)`. From a templated API point of view there is no need to distinguish to distinguish between single sample or bulk operation. The implementation is different (for efficiency reasons) but the API should stay the same. 

`push_back` and `push_front' keep the symmetry between reverse (nominal) and forward (new optional) behaviour.

Reply:
> I didn't want to break existing code, thus keeping the "old" push_back_bulk(...).

I was hoping we're still in the 'we can break API things' as we didn't have a release. (and we do break behaviour in other places from time to time)